packages/organization/src/organization.ts

```diff
@@ -0,0 +1,148 @@
+};
+
+export const switchTierHelper = async (
+  organizationId: string,
+  newTier: Tier,
+  query: Omit<Prisma.OrganizationFindUniqueArgs, 'where'>,
+): Promise<Organization> => {
+  const disconnectIntegration = async (maxIntegrations: number, orgId: string): Promise<void> => {
+    const integrations = await prisma.integration.findMany({
+      where: {
+        organizationId: orgId,
+      },
+      orderBy: {
+        createdAt: 'asc',
+      },
+    });
+
+    if (integrations.length > maxIntegrations) {
+      const integrationsToDisconnect = integrations.slice(maxIntegrations);
+
+      integrationsToDisconnect.forEach((integration) => {
+        fireAndForget.add(() => revokeIntegration(integration.externalId, integration.type));
+      });
+    }
+  };
+  const removeUserFromOrganization = async (
+    orgId: string,
+    allUsers: OrganizationWithUsers[`users`],
+    tier: Tier,
+    subOrganizations: OrganizationWithUsers[],
+  ): Promise<void> => {
+    const maxUsersInTier = tierConstraints[tier].maxUsers;
+    if (tier !== Tier.Launch && tier !== Tier.Build && tier !== Tier.Grow) {
+      return;
+    }
+
+    const sortedUsers = Array.from(new Set(allUsers)).sort(
+      (a, b) => new Date(a.user.createdAt).getTime() - new Date(b.user.createdAt).getTime(),
+    );
+
+    const usersToKeep = sortedUsers.slice(0, maxUsersInTier);
+
+    const adminInKeptUsers = usersToKeep.some((user) => user.role === OrganizationRoleEnum.ORG_ADMIN);
+
+    if (!adminInKeptUsers) {
+      const oldestAdmin = sortedUsers.find((user) => user.role === OrganizationRoleEnum.ORG_ADMIN);
+      if (oldestAdmin) {
+        usersToKeep.pop();
+        usersToKeep.push(oldestAdmin);
+      }
+    }
+
+    const usersToRemove = sortedUsers.filter((user) => !usersToKeep.some((keep) => keep.userId === user.userId));
+    const subOrgIds = subOrganizations.map((subOrg) => subOrg.id);
+    const organizationIds = [orgId, ...subOrgIds];
+
+    await prisma.userOrganization.deleteMany({
+      where: {
+        userId: { in: usersToRemove.map((userOrg) => userOrg.userId) },
+        organizationId: { in: organizationIds },
+      },
+    });
+
+    for (const subOrg of subOrganizations) {
+      const subOrgAdmin = Array.from(new Set(subOrg.users))
+        .sort((a, b) => new Date(a.user.createdAt).getTime() - new Date(b.user.createdAt).getTime())
+        .find((user) => user.role === OrganizationRoleEnum.ORG_ADMIN);
+      if (!subOrgAdmin) {
+        const rootOrgAdmin = usersToKeep
+          .sort((a, b) => new Date(a.user.createdAt).getTime() - new Date(b.user.createdAt).getTime())
+          .find((user) => user.role === OrganizationRoleEnum.ORG_ADMIN);
+
+        if (rootOrgAdmin) {
+          await prisma.userOrganization.update({
+            where: { userId_organizationId: { userId: rootOrgAdmin.userId, organizationId: subOrg.id } },
+            data: { role: OrganizationRoleEnum.ORG_ADMIN },
+          });
+        }
+      }
+    }
+  };
```

Comment:
```suggestion
    for (const org of subOrganizations) {
      const keptUsers = usersToKeep.filter((user) => user.organizationId === org.id);
      if (!keptUsers.some((u) => u.role === OrganizationRoleEnum.ORG_ADMIN) && oldestAdmin) {
        await prisma.userOrganization.update({
          where: { userId_organizationId: { userId: oldestAdmin.userId, organizationId: org.id } },
          data: { role: OrganizationRoleEnum.ORG_ADMIN },
        });
      }
    }
```
do you see any bug here? You will need to move oldestOrganization outside the if.